src/MessageContainer.js

```diff
@@ -135,10 +135,20 @@ export default class MessageContainer extends React.Component {
     );
   }
 
+  endReachedProps(){
+    let condProps = {}
+    if (this.props.onTopReached){ condProps.onEndReached = this.props.onTopReached }
```

Comment:
Hi @dgdavid and @kfiroo ,
The `InvertibleScrollView` receives `onEndReached` as `ScrollView`,  but the Y axis is inverted so it triggers `endReached` when is at the top.
On scroll is not as reliable as `end/topReached` reached is only triggered once onScroll is more unpredictable.